aiida/orm/nodes/data/singlefile.py

```diff
@@ -56,17 +57,19 @@ def filename(self):
         """
         return self.get_attribute('filename')
 
-    def open(self, key=None, mode='r'):
+    @contextlib.contextmanager
+    def open(self, path=None, mode='r'):
         """Return an open file handle to the content of this data node.
 
-        :param key: optional key within the repository, by default is the `filename` set in the attributes
+        :param path: the relative path of the object within the repository.
```

Comment:
Renamed `key` -> `path`. In other places a deprecation warning is added for this.

Reply:
Thanks, will add to `UpfData` and `CifData` subclasses as well if necessary